developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,133 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID Needed
+***************
+
+To develope a new audio signal processing component, we traditionally
+implement the component driver with a new unique component type
+introduced, the painpoints with this method including how to keep ABI
+aligned and backward compatible for different version combination of
+topology/driver/firmware for each new added components, there could be
+component type collision if versions are not aligned.
+
+UUIDs (Universally Unique Identifiers) provide a more scalable and
+collision-free way of component identification. UUIDs are currently
+used as the standard interface by all users of the SOF firmware,
+including the tracing subsystem, the topology .m4 files, and the Linux
+topology driver.
+
+Allocating a new UUID for the new added component is recommended, as
+the component type might be replaced by UUID in the future. For the
+whole SOF stack, the usage of UUID should follow rules as below:
+
+UUID Allocation
+***************
+The UUID allocation of a specific component should use the version 4 in
```

Comment:
shall use. It's a mandatory requirement. should is a recommendation.

Reply:
Good learning, thanks.